### PR TITLE
rpc: return transfers for migrated contracts

### LINKED ISSUE #1215: getnep5transfers shows nothing in received although there should be something
```
$ curl -d '{"jsonrpc": "2.0", "method": "getnep5transfers", "params": ["AKinx3r8YPQRxS6WyLEGEQvcNNAEkto3Ek"], "id": 1}' https://rpc4.go.nspcc.ru:10331 | json_pp
{
   "jsonrpc" : "2.0",
   "id" : 1,
   "result" : {
      "received" : [],
      "sent" : [
         {
            "block_index" : 5791359,
            "transfer_notify_index" : 0,
            "timestamp" : 1593621641,
            "tx_hash" : "0xbd8a55734664f4ac7ed220e7b47511de2040ebefda9d973e0cee07de3fa98e48",
            "asset_hash" : "0x53b7577befb37d4d3b95a02f60f5da8933ab5f04",
            "transfer_address" : "Ack4DSNYjFqQcgkAVCyiotCHVLkNdkLk5Y",
            "amount" : "0.00150000"
         },
         {
            "transfer_address" : "ATAiwLv2XQEyDXFF6zHCVKHtKkAT7ksQMT",
            "amount" : "2.99850000",
            "timestamp" : 1593621641,
            "transfer_notify_index" : 0,
            "block_index" : 5791359,
            "tx_hash" : "0xbd8a55734664f4ac7ed220e7b47511de2040ebefda9d973e0cee07de3fa98e48",
            "asset_hash" : "0x53b7577befb37d4d3b95a02f60f5da8933ab5f04"
         },
         {
            "asset_hash" : "0x53b7577befb37d4d3b95a02f60f5da8933ab5f04",
            "tx_hash" : "0x70e644ab86780aa9ba63ee07933a7deaba81efac3918e402964cb38dce800f9a",
            "transfer_notify_index" : 0,
            "block_index" : 5800383,
            "timestamp" : 1593777377,
            "amount" : "1",
            "transfer_address" : "AFwYT3HDAwkTneE8JytrCG1MWAFaqBUmnr"
         },
         {
            "tx_hash" : "0xcdd46a0acbcf9299e44fcc029f35fe5eacb0bfec6518cd3dce85503048bc2171",
            "asset_hash" : "0x53b7577befb37d4d3b95a02f60f5da8933ab5f04",
            "transfer_notify_index" : 0,
            "block_index" : 5800505,
            "timestamp" : 1593779445,
            "amount" : "5",
            "transfer_address" : "AFwYT3HDAwkTneE8JytrCG1MWAFaqBUmnr"
         }
      ],
      "address" : "AKinx3r8YPQRxS6WyLEGEQvcNNAEkto3Ek"
   }
}
```
So there is nothing in `received` although there should be something, otherwise how could it send anything?

### PR DESCRIPTION
**Pull request by fyrchik**
Transfers for migrated contracts are still saved in the transfer log, but we have no way to get decimals.
We can either save decimals info elsewhere or detect if contract was migrated.
This PR implements second solution as it is simpler and backwards compatible.

Fix #1215 . However, I still need to restore mainnet to 6m block to be really sure :)

### CODE REVIEW

File: pkg/core/interop_neo.go [hunk 1]
```diff
@@ -599,6 +600,10 @@ func (ic *interopContext) contractMigrate(v *vm.VM) error {
 			}
 			ic.dao.MigrateNEP5Balances(hash, contract.ScriptHash())
 		}
+		err = ic.dao.PutMigratedAddress(hash, contract.ScriptHash())
```
Comment (fyrchik) [inline pkg/core/interop_neo.go:603]: We can save tx hash also, but I haven't found usecases other than fixing bugs.

**Comment by roman-khimov** [comment]
This contract hash map is something that I've specifically tried to avoid with #1144 fix. And this approach is a bit more dangerous here, because the contract might be deleted, not migrated. How are we going to answer these calls for deleted contracts?

**Comment by fyrchik** [comment]
@roman-khimov transfers for destroyed contracts won't be displayed.
I was thinking about other ways to fix:
1. Saving strings in transfer log. This requires querying decimals in `storeBlock` which is quite a big overhead imo. But it will work for destroyed contracts.
2. Saving "real hash" in transfer log and updating it upon migration.

**Comment by roman-khimov** [comment]
How about maintaining a map of contract hashes to NEP5 metadata (we may start with saving only decimals)? We can fill it when migrating or destroying a contract.

**Comment by fyrchik** [comment]
Tested on maiinnet, works as expected.

**Pull request merged**